examples/light-switch-app/efr32/BUILD.gn

```diff
@@ -199,20 +199,12 @@ efr32_executable("light_switch_app") {
   # OpenThread Settings
   if (chip_enable_openthread) {
     deps += [
-      "${chip_root}/third_party/openthread/platforms:libopenthread-platform",
-      "${chip_root}/third_party/openthread/platforms:libopenthread-platform-utils",
+      "${chip_root}/third_party/openthread:openthread_device_deps",
       "${examples_plat_dir}:efr-matter-shell",
     ]
-    if (chip_openthread_ftd) {
-      deps += [
-        "${chip_root}/third_party/openthread/repo:libopenthread-cli-ftd",
-        "${chip_root}/third_party/openthread/repo:libopenthread-ftd",
-      ]
-    } else {
-      deps += [
-        "${chip_root}/third_party/openthread/repo:libopenthread-cli-mtd",
-        "${chip_root}/third_party/openthread/repo:libopenthread-mtd",
-      ]
+
+    if (enable_openthread_cli) {
+      deps += [ "${efr32_sdk_build_root}:openthread_cli" ]
```

Comment:
This seems misplaced here. The code using this is in GenericThreadStackManagerImpl_OpenThread.cpp, correct?

So the dependency on libopenthread-cli should be in the platform library, and most likely it would openthread-cli management should also be centralized in third_party/openthread/BUILD.gn

Reply:
the whole example can use thread cli. It is also usable for matter shell if you enable this.

It can probably be centralized yes. But doing this can affect other platforms as they enable it in their examples similarly to how it is done here.

I rather do that step in a separate PR

Reply:
Does the example code invoke the CLI directly?

The usual rule is that only direct dependencies should be listed, so you only list a library if you include one of its header.

In this directory I see the following includes:

```
#include "AppConfig.h"
#include "AppEvent.h"
#include "AppTask.h"
#include "BaseApplication.h"
#include "FreeRTOS.h"
#include "LEDWidget.h"
#include "app-common/zap-generated/ids/Clusters.h"
#include "app-common/zap-generated/ids/Commands.h"
#include "app/CommandSender.h"
#include "app/clusters/bindings/BindingManager.h"
#include "app/server/Server.h"
#include "binding-handler.h"
#include "controller/InvokeInteraction.h"
#include "init_efrPlatform.h"
#include "lcd.h"
#include "lib/core/CHIPError.h"
#include "lib/shell/Engine.h"
#include "lib/shell/commands/Help.h"
#include "platform/CHIPDeviceLayer.h"
#include "qrcodegen.h"
#include "sl_simple_button_instances.h"
#include "sl_simple_led_instances.h"
#include "sl_system_kernel.h"
#include "timers.h" // provides FreeRTOS timer support
#include <AppTask.h>
#include <DeviceInfoProviderImpl.h>
#include <app-common/zap-generated/attribute-id.h>
#include <app-common/zap-generated/attribute-type.h>
#include <app-common/zap-generated/cluster-id.h>
#include <app-common/zap-generated/ids/Attributes.h>
#include <app-common/zap-generated/ids/Clusters.h>
#include <app/ConcreteAttributePath.h>
#include <app/clusters/bindings/bindings.h>
#include <app/clusters/identify-server/identify-server.h>
#include <app/server/OnboardingCodesUtil.h>
#include <app/server/Server.h>
#include <app/util/attribute-storage.h>
#include <assert.h>
#include <ble/BLEEndPoint.h>
#include <credentials/DeviceAttestationCredsProvider.h>
#include <credentials/examples/DeviceAttestationCredsExample.h>
#include <examples/platform/efr32/EFR32DeviceAttestationCreds.h>
#include <lib/core/CHIPError.h>
#include <lib/support/CodeUtils.h>
#include <lib/support/logging/CHIPLogging.h>
#include <matter_config.h>
#include <platform/CHIPDeviceLayer.h>
#include <platform/EFR32/freertos_bluetooth.h>
#include <setup_payload/QRCodeSetupPayloadGenerator.h>
#include <setup_payload/SetupPayload.h>
#include <stdbool.h>
#include <stdint.h>
```

If none of those are openthread CLI (which?) it should not be listed.

Agree it's a preexisting issue but generally it's much easier to understand your changes if the cleanup happens before the new functionality is added.

Reply:
you are right it is only used in GenericThreadStackManagerImpl_OpenThread and src/platform/XXX/ThreadStackManagerImpl.cpp

Reply:
move the deps out of the examples buid.gn to single location the src/platform/efr32/build.gn

I will make a second PR this week to move the dependency definition to `third_party/openthread/BUILD.gn` and deal with that change for other platforms then